lib/pkcs11h-core.c

```diff
@@ -1345,3 +1656,97 @@ __pkcs11h_forkFixup () {
 
 #endif				/* !WIN32 */
 
+static
+_pkcs11h_provider_t
+__pkcs11h_get_pkcs11_provider(const char * const reference) {
+	_pkcs11h_provider_t provider = NULL;
+
+	provider = _g_pkcs11h_data->providers;
+	while (
+		provider != NULL &&
+		strcmp (reference, provider->reference)
+	) {
+		provider = provider->next;
+	}
+
+	return provider;
+}
+
+static
+const char*
+__pkcs11h_property_type_to_str(const pkcs11h_provider_property_type_t property) {
+	switch (property) {
+		case PKCS11H_PROVIDER_PROPERTY_LOCATION:
+			return "PKCS11H_PROVIDER_PROPERTY_LOCATION";
```

Comment:
if you declare this as macro you can use \#XXX, if not, please use an array of strings.